storage/repository/parse.go

```diff
@@ -44,14 +45,20 @@ func NewFileURI(path string) fyne.URI {
 func ParseURI(s string) (fyne.URI, error) {
 	// Extract the scheme.
 	scheme := ""
+	validScheme := false
 	for i := 0; i < len(s); i++ {
 		if s[i] == ':' {
+			validScheme = true
 			break
 		}
 		scheme += string(s[i])
 	}
 	scheme = strings.ToLower(scheme)
 
+	if !validScheme {
+		return nil, errors.New("invalid URI, scheme must be present")
+	}
+
```

Comment:
The previous code here actually looks rather inefficient and not that good. I am thinking that it would be better to just replace the whole loop with something like this:
```go
colonIndex := strings.Index(s, "://")
if colonIndex == -1 {
     return nil, errors.New("invalid URI, scheme must be present")
}

scheme := s[:colonIndex]
```
This should be cleaner and a lot better than performing string concatenation all the time.

Reply:
Good call. It appears that this code also needs to support URNs (storage/uri_test.go:26) so I've implemented your suggestion however just limiting it to the colon.